### PR TITLE
[Merged by Bors] - feat: port GroupTheory.GroupAction.SubMulAction.Pointwise

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
discussion on instances' names here: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/GroupTheory.2EGroupAction.2ESubMulAction.2EPointwise

File: Mathlib/GroupTheory/GroupAction/SubMulAction/Pointwise.lean [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+theorem coe_pow (p : SubMulAction R M) : ∀ {n : ℕ} (_ : n ≠ 0), ↑(p ^ n) = (p ^ n : Set M)
+  | 0, hn => (hn rfl).elim
+  | 1, _ => by rw [pow_one]
+  | n + 2, _ => by rw [pow_succ _ (n + 1)]
+#align sub_mul_action.coe_pow SubMulAction.coe_pow
+
+theorem subset_coe_pow (p : SubMulAction R M) : ∀ {n : ℕ}, (p ^ n : Set M) ⊆ ↑(p ^ n)
+  | 0 => by rw [pow_zero]
+  | n + 1 => (coe_pow p n.succ_ne_zero).superset
+#align sub_mul_action.subset_coe_pow SubMulAction.subset_coe_pow
```
Comment (mo271) [inline Mathlib/GroupTheory/GroupAction/SubMulAction/Pointwise.lean:132]: are those lemmas needed in mathlib4?

File: Mathlib/GroupTheory/GroupAction/SubMulAction/Pointwise.lean [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  | n + 2, _ => by
+    rw [pow_succ _ (n + 1), pow_succ _ (n + 1), coe_mul, coe_pow _ n.succ_ne_zero]
+#align sub_mul_action.coe_pow SubMulAction.coe_pow
+
+theorem subset_coe_pow (p : SubMulAction R M) : ∀ {n : ℕ}, ((p : Set M) ^ n) ⊆ (p ^ n : Set M)
+  | 0 => by
+    rw [pow_zero, pow_zero]
+    exact subset_coe_one
+  | n + 1 => by rw [← Nat.succ_eq_add_one, coe_pow _ n.succ_ne_zero]
+#align sub_mul_action.subset_coe_pow SubMulAction.subset_coe_pow
```
Comment (mo271) [inline Mathlib/GroupTheory/GroupAction/SubMulAction/Pointwise.lean:135]: Not entirely sure if I did the right thing here with the coercions.

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**